src/main/java/it/gov/pagopa/cgn/portal/service/DiscountService.java

```diff
@@ -248,19 +287,28 @@ private ProfileEntity validateDiscount(String agreementId, DiscountEntity discou
             discountEntity.setLandingPageUrl(null);
             discountEntity.setLandingPageReferrer(null);
             discountEntity.setLastBucketCodeLoadUid(null);
+            discountEntity.setLastBucketCodeLoadFileName(null);
+            discountEntity.setLastBucketCodeLoad(null);
+            bucketService.deleteBucketCodes(discountEntity.getId());
```

Comment:
maybe this operation could take long time if there are millions of codes to delete. Are we sure that we want to delete orphans sincronously?

Reply:
We could also call the async method, you're right, we have it.